starport_template/lib/widgets/send_money_form.dart

```diff
@@ -23,24 +29,25 @@ class SendMoneyForm extends StatelessWidget {
       child: Column(
         children: [
           CosmosTextField(
-            onChanged: (value) => onAddressChanged(value),
+            onChanged: (value) => widget.onAddressChanged(value),
+            text: text,
             hint: "Enter receiver's wallet address",
-            suffix: CosmosTextButton(onTap: () {}, text: 'Paste', color: CosmosTheme.of(context).colors.link),
+            suffix: CosmosTextButton(
+              onTap: _onTapPaste,
+              text: 'Paste',
+              color: CosmosTheme.of(context).colors.link,
+            ),
           ),
           SizedBox(height: theme.spacingL),
           CosmosTextField(
-            onChanged: (value) => onAmountChanged(value),
+            onChanged: (value) => widget.onAmountChanged(value),
             keyboardType: TextInputType.number,
-            hint: '0 ${denomText.toUpperCase()}',
-          ),
-          SizedBox(height: theme.spacingL),
-          CosmosTextField(
-            onChanged: (value) => onFeeChanged(value),
-            keyboardType: TextInputType.number,
-            hint: 'Transaction fee',
+            hint: '0 ${widget.denomText.toUpperCase()}',
           ),
         ],
       ),
     );
   }
+
+  Future<void> _onTapPaste() => FlutterClipboard.paste().then((value) => setState(() => text = value));
```

Comment:
how about we use the normal async/await flow here:
```suggestion
  Future<void> _onTapPaste() {
    final value = await FlutterClipboard.paste();
    setState(() => text = value);
}
```